pkg/services/ngalert/notifier/channels/slack.go

```diff
@@ -35,132 +35,110 @@ type SlackNotifier struct {
 	tmpl          *template.Template
 	images        ImageStore
 	webhookSender notifications.WebhookSender
-
-	URL            *url.URL
-	Username       string
-	IconEmoji      string
-	IconURL        string
-	Recipient      string
-	Text           string
-	Title          string
-	MentionUsers   []string
-	MentionGroups  []string
-	MentionChannel string
-	Token          string
+	settings      slackSettings
 }
 
-type SlackConfig struct {
-	*NotificationChannelConfig
-	URL            *url.URL
-	Username       string
-	IconEmoji      string
-	IconURL        string
-	Recipient      string
-	Text           string
-	Title          string
-	MentionUsers   []string
-	MentionGroups  []string
-	MentionChannel string
-	Token          string
+type slackSettings struct {
+	EndpointURL      string   `json:"endpointUrl,omitempty" yaml:"endpointUrl,omitempty"`
+	URL              string   `json:"url,omitempty" yaml:"url,omitempty"`
+	Token            string   `json:"token,omitempty" yaml:"token,omitempty"`
+	Recipient        string   `json:"recipient,omitempty" yaml:"recipient,omitempty"`
+	Text             string   `json:"text,omitempty" yaml:"text,omitempty"`
+	Title            string   `json:"title,omitempty" yaml:"title,omitempty"`
+	Username         string   `json:"username,omitempty" yaml:"username,omitempty"`
+	IconEmoji        string   `json:"icon_emoji,omitempty" yaml:"icon_emoji,omitempty"`
+	IconURL          string   `json:"icon_url,omitempty" yaml:"icon_url,omitempty"`
+	MentionChannel   string   `json:"mentionChannel,omitempty" yaml:"mentionChannel,omitempty"`
+	MentionUsersStr  string   `json:"mentionUsers,omitempty" yaml:"mentionUsers,omitempty"`
+	MentionUsers     []string `json:"-" yaml:"-"`
+	MentionGroupsStr string   `json:"mentionGroups,omitempty" yaml:"mentionGroups,omitempty"`
+	MentionGroups    []string `json:"-" yaml:"-"`
 }
 
+// SlackFactory creates a new NotificationChannel that sends notifications to Slack.
 func SlackFactory(fc FactoryConfig) (NotificationChannel, error) {
-	cfg, err := NewSlackConfig(fc)
+	ch, err := buildSlackNotifier(fc)
 	if err != nil {
 		return nil, receiverInitError{
 			Reason: err.Error(),
 			Cfg:    *fc.Config,
 		}
 	}
-	return NewSlackNotifier(cfg, fc.ImageStore, fc.NotificationService, fc.Template), nil
+	return ch, nil
 }
 
-func NewSlackConfig(factoryConfig FactoryConfig) (*SlackConfig, error) {
+func buildSlackNotifier(factoryConfig FactoryConfig) (*SlackNotifier, error) {
 	channelConfig := factoryConfig.Config
 	decryptFunc := factoryConfig.DecryptFunc
-	endpointURL := channelConfig.Settings.Get("endpointUrl").MustString(SlackAPIEndpoint)
-	slackURL := decryptFunc(context.Background(), channelConfig.SecureSettings, "url", channelConfig.Settings.Get("url").MustString())
+	var settings slackSettings
+	err := factoryConfig.Config.marshalSettings(&settings)
```

Comment:
Maybe change this to `unmarshalSettings`? We're unmarshalling JSON into a struct.